### PR TITLE
chore: dep updates and minor internal type fixes

### PR DESCRIPTION
**Pull request by rchl**
Nothing important really. Just felt like enabling type checking for *.vue and *.js files to catch some extra issues.

### CODE REVIEW

**Comment by rchl** [comment]
Windows and Ubuntu seem to timeout on running `test:e2e-generated`. Not quite sure what's the reason. Even tried locally on a Windows virtual machine and it works fine there.

```
Run yarn test:e2e-generated
yarn run v1.22.19
$ cross-env GENERATE=true PORT=5000 start-server-and-test fixture:generate 5000 "testcafe -q firefox:headless test/e2e"
1: starting server using command "npm run fixture:generate"
and when url "[ 'http://localhost:5000' ]" is responding with HTTP status code NaN
running tests using command "testcafe -q firefox:headless test/e2e"


> @nuxtjs/composition-api@0.33.1 fixture:generate
> yarn fixture:generate:export && yarn http-server -s -p 5000 test/fixture/dist

$ yarn clean:fixture && cross-env GENERATE=true PORT=6000 CMD=generate yarn nuxt-run
$ rimraf test/fixture/dist/ test/fixture/.nuxt
$ cross-env-shell "yarn nuxt $CMD test/fixture"
$ /home/runner/work/composition-api/composition-api/node_modules/.bin/nuxt generate test/fixture
[log] Testing built module
[log] Testing built module
$ /home/runner/work/composition-api/composition-api/node_modules/.bin/http-server -s -p 5000 test/fixture/dist
Error: Timed out waiting for: http://localhost:5000
    at /home/runner/work/composition-api/composition-api/node_modules/wait-on/lib/wait-on.js:132:31
    at doInnerSub (/home/runner/work/composition-api/composition-api/node_modules/wait-on/node_modules/rxjs/dist/cjs/internal/operators/mergeInternals.js:22:31)
    at outerNext (/home/runner/work/composition-api/composition-api/node_modules/wait-on/node_modules/rxjs/dist/cjs/internal/operators/mergeInternals.js:17:70)
    at OperatorSubscriber._this._next (/home/runner/work/composition-api/composition-api/node_modules/wait-on/node_modules/rxjs/dist/cjs/internal/operators/OperatorSubscriber.js:33:21)
    at Subscriber.next (/home/runner/work/composition-api/composition-api/node_modules/wait-on/node_modules/rxjs/dist/cjs/internal/Subscriber.js:51:18)
    at AsyncAction.work (/home/runner/work/composition-api/composition-api/node_modules/wait-on/node_modules/rxjs/dist/cjs/internal/observable/timer.js:28:28)
    at AsyncAction._execute (/home/runner/work/composition-api/composition-api/node_modules/wait-on/node_modules/rxjs/dist/cjs/internal/scheduler/AsyncAction.js:79:18)
    at AsyncAction.execute (/home/runner/work/composition-api/composition-api/node_modules/wait-on/node_modules/rxjs/dist/cjs/internal/scheduler/AsyncAction.js:67:26)
    at AsyncScheduler.flush (/home/runner/work/composition-api/composition-api/node_modules/wait-on/node_modules/rxjs/dist/cjs/internal/scheduler/AsyncScheduler.js:38:33)
    at listOnTimeout (node:internal/timers:569:17)
    at process.processTimers (node:internal/timers:512:7)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Process completed with exit code 1.
```

**Comment by danielroe** [comment]
Merged https://github.com/nuxt-community/composition-api/pull/740 to address CI failure - hopefully things will work once rebased/merged.

**Comment by rchl** [comment]
Switched to Node 16? Should have thought of that :)

**Comment by rchl** [comment]
Looking good now.

File: test/fixture/nuxt.config.ts [hunk 1]
```diff
@@ -15,6 +15,7 @@ const moduleSource = join(rootDir, inDevelopment ? 'src' : 'dist')
 console.log('Testing', inDevelopment ? 'source' : 'built', 'module')
 
 export default <NuxtConfig>{
+  telemetry: false,
```
Comment (danielroe) [inline test/fixture/nuxt.config.ts:18]: this should be disabled in test environments anyway

Comment (danielroe) [reply test/fixture/nuxt.config.ts:18]:
```suggestion
```

**Review by danielroe — APPROVED** [review]

File: test/fixture/nuxt.config.ts (hunk 1 above)
Comment (rchl) [reply test/fixture/nuxt.config.ts]: It is but it's not disabled when running tests locally. At least in some of the cases, i can check which one exactly later

Comment (rchl) [reply test/fixture/nuxt.config.ts]: I can't reproduce this anymore so I'll revert this change.
Maybe update of dependencies made it irrelevant.

Comment (rchl) [reply test/fixture/nuxt.config.ts]: Oh, you've removed it already. Good stuff.

**Pull request merged**